packages/contracts-bedrock/contracts/deployment/FreshSystemDictator.sol

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { ProxyAdmin } from "../universal/ProxyAdmin.sol";
+import { L2OutputOracle } from "../L1/L2OutputOracle.sol";
+import { OptimismPortal } from "../L1/OptimismPortal.sol";
+import { L1CrossDomainMessenger } from "../L1/L1CrossDomainMessenger.sol";
+import { L1StandardBridge } from "../L1/L1StandardBridge.sol";
+import { OptimismMintableERC20Factory } from "../universal/OptimismMintableERC20Factory.sol";
+
+/**
+ * @title FreshSystemDictator
+ * @notice The FreshSystemDictator is responsible for coordinating initialization of a fresh
+ *         deployment of the Optimism system. We expect that all proxies and implementations
+ *         already be deployed before this contract is used.
+ */
+contract FreshSystemDictator is Ownable {
+    struct GlobalConfig {
+        ProxyAdmin proxyAdmin;
+        address controller;
+        address finalOwner;
+    }
+
+    struct ProxyAddressConfig {
+        address l2OutputOracleProxy;
+        address optimismPortalProxy;
+        address l1CrossDomainMessengerProxy;
+        address l1StandardBridgeProxy;
+        address optimismMintableERC20FactoryProxy;
+    }
+
+    struct ImplementationAddressConfig {
+        L2OutputOracle l2OutputOracleImpl;
+        OptimismPortal optimismPortalImpl;
+        L1CrossDomainMessenger l1CrossDomainMessengerImpl;
+        L1StandardBridge l1StandardBridgeImpl;
+        OptimismMintableERC20Factory optimismMintableERC20FactoryImpl;
+    }
+
+    struct L2OutputOracleConfig {
+        bytes32 l2OutputOracleGenesisL2Output;
+        address l2OutputOracleProposer;
+        address l2OutputOracleOwner;
+    }
+
+    struct Config {
+        GlobalConfig globalConfig;
+        ProxyAddressConfig proxyAddressConfig;
+        ImplementationAddressConfig implementationAddressConfig;
+        L2OutputOracleConfig l2OutputOracleConfig;
+    }
+
+    Config public config;
+
+    constructor(Config memory _config) Ownable() {
+        config = _config;
+        _transferOwnership(config.globalConfig.controller);
+    }
+
+    function step1() external onlyOwner {
```

Comment:
I believe we do not need to upgrade the L1ERC721Bridge as part of this, just mentioning it to double check